Sourcery/Parsing/Parser.swift

```diff
@@ -453,18 +456,57 @@ extension Parser {
 
         return annotations
     }
+    
+    fileprivate func parseTypealiases(from source: [String: SourceKitRepresentable], containingType: Type?, processed: [[String: SourceKitRepresentable]]) -> [Typealias] {
+        var contents = self.contents
+        
+        //replace all processed substructures with whitespaces so that we don't process their typealiases again
+        for substructure in processed {
+            if let offset = substructure[SwiftDocKey.offset.rawValue] as? Int64,
+                let length = substructure[SwiftDocKey.length.rawValue] as? Int64,
+                let nsRange = contents.byteRangeToNSRange(start: Int(offset), length: Int(length)) {
+                
+                let replacement = String(repeating: " ", count: nsRange.length)
+                contents = contents.bridge().replacingCharacters(in: nsRange, with: replacement)
+            }
+        }
+        if containingType != nil {
+            if let body = extract(.body, from: source, contents: contents) {
+                return parseTypealiases(SyntaxMap(file: File(contents: body)).tokens, contents: body)
+            } else {
+                return []
+            }
+        } else {
+            return parseTypealiases(SyntaxMap(file: File(contents: contents)).tokens, contents: contents)
+        }
+    }
+    
+    private func parseTypealiases(_ tokens: [SyntaxToken], contents: String, existingTypealiases: [Typealias] = []) -> [Typealias] {
+        var typealiases = existingTypealiases
+        tokens.enumerated()
+            .forEach { index, token in
+                if token.type == "source.lang.swift.syntaxtype.keyword",
+                    extract(token, contents: contents) == "typealias",
+                    let alias = extract(tokens[index + 1], contents: contents),
+                    let type = extract(tokens[index + 2], contents: contents) {
+                    
+                    typealiases.append(Typealias(aliasName: alias, typeName: type))
+                }
+        }
+        return typealiases
+    }
 
     fileprivate func isGeneric(source: [String: SourceKitRepresentable]) -> Bool {
-        guard let substring = extract(.nameSuffix, from: source), substring.trimmingCharacters(in: .whitespacesAndNewlines).hasPrefix("<") == true else { return false }
+        guard let substring = extract(.nameSuffix, from: source, contents: contents), substring.trimmingCharacters(in: .whitespacesAndNewlines).hasPrefix("<") == true else { return false }
         return true
     }
 
-    fileprivate func extract(_ substringIdentifier: SubstringIdentifier, from source: [String: SourceKitRepresentable]) -> String? {
-        let substring = substringIdentifier.range(for: source).flatMap { self.contents.substringWithByteRange(start: Int($0.offset), length: Int($0.length)) }
+    fileprivate func extract(_ substringIdentifier: SubstringIdentifier, from source: [String: SourceKitRepresentable], contents: String) -> String? {
```

Comment:
why always pass contents in? add a convenience variant that uses this function but passes in self.contents. That way you only use a subset when explicitly needed